tests/planner_test/tpch.planner.sql

```diff
@@ -112,12 +112,59 @@ order by
     l_linestatus;
 
 /*
-Projection: [l_returnflag, l_linestatus, sum(l_quantity), sum(l_extendedprice), sum((l_extendedprice * (1 - l_discount))), sum(((l_tax + 1) * (l_extendedprice * (1 - l_discount)))), (sum(l_quantity) / count(l_quantity)), (sum(l_extendedprice) / count(l_extendedprice)), (sum(l_discount) / count(l_discount)), rowcount] (cost=39238.984)
-  Order: [l_returnflag asc, l_linestatus asc] (cost=33158.984)
-    Aggregate: [sum(l_quantity), sum(l_extendedprice), sum((l_extendedprice * (1 - l_discount))), sum(((l_tax + 1) * (l_extendedprice * (1 - l_discount)))), count(l_quantity), count(l_extendedprice), sum(l_discount), count(l_discount), rowcount], groupby=[l_returnflag, l_linestatus] (cost=25300)
-      Projection: [l_quantity, l_extendedprice, l_discount, l_tax, l_returnflag, l_linestatus] (cost=18260)
-        Filter: (1998-09-21 >= l_shipdate) (cost=12900)
-          Scan: lineitem[l_quantity, l_extendedprice, l_discount, l_tax, l_returnflag, l_linestatus, l_shipdate] (cost=7000)
+
+Projection                                                                                                              
+├── cost: 39238.984                                                                                                     
+├── exprs:                                                                                                              
+│   ┌── l_returnflag                                                                                                    
+│   ├── l_linestatus                                                                                                    
+│   ├── sum {  }                                                                                                        
+│   │   └── l_quantity                                                                                                  
+│   ├── sum {  }                                                                                                        
+│   │   └── l_extendedprice                                                                                             
+│   ├── sum {  }                                                                                                        
+│   │   └── * { lhs: l_extendedprice, rhs: - { lhs: 1, rhs: l_discount } }                                              
+│   ├── sum {  }                                                                                                        
+│   │   └── * { lhs: + { lhs: l_tax, rhs: 1 }, rhs: * { lhs: l_extendedprice, rhs: - { lhs: 1, rhs: l_discount } } }    
```

Comment:
`lhs` and `rhs` looks verbose in expressions. How about this:
```suggestion
│   │   └── *(+(l_tax, 1), *(l_extendedprice, -(1, l_discount)))   
```

Reply:
It is very hard to make a special case for binary expressions. You'll have to change the generator to work for binary expressions in general.

Reply:
A feature request for pretty-xmlish. 🤓

Reply:
How to handle BinExpr with enode = +, lhs = a scan node, rhs = an insert node? Or may I assume that such expressions will never appear?

What is the criteria for expressions to be valid for risinglight?

Reply:
We should assume that will never happen, but I haven't find a proper way to ensure that in the egg framework.
In risinglight, expressions are e-node trees without any plan nodes. Though there's no formal definition yet. 😅

Reply:
I strongly believe that there should be a separate pretty printer for expressions. I know how to do it (and it's done properly in the Aya Prover), but it requires changing the AST structure.